### PR TITLE
Move management clients into management package

### PR DESCRIPTION
**Pull request by jen20**
This commit moves around the packages for the Azure SDK as discussed in #45.

Management and Storage have been moved to the top level of packages along with Core.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,3 +1,3 @@
 language: go
 go: tip
-script: go test -v ./azure/...
+script: go test -v ./storage/...
```
Comment (ahmetalpbalkan) [inline .travis.yml]: we can add an extra step like `go build ./mangement/...` to check if it builds.

**Comment by ahmetalpbalkan** [comment]
LGTM.

I can probably split blob storage client to storage/blob/ sometime later. It's not really urgent and requires a bit of extra work.

**Comment by ahmetalpbalkan** [comment]
@jen20 I really hate Makefiles personally and don't use them except really necessary. The `go` command pretty much aims to eliminate a need for Makefiles. Is there a particular reason you incorporated makefiles in this change? circle.yml file does support multiple commands very well.

**Comment by jen20** [comment]
@ahmetalpbalkan it's just the easiest most cross platform (Windows excepted, though it's even there) way of running all this stuff prior to checkin. Also, it's using travis rather than circle, but there's still no trivial way to run everything locally with one command to even check builds prior to push.

**Comment by ahmetalpbalkan** [comment]
@jen20 yep, as you said it requires make on windows :-) I think we should undo that commit and discuss part of another PR. My argument mostly evolves around motivation for "go" command and why they thought Makefile shouldn't exist for Go projects here: https://golang.org/doc/articles/go_command.html

**Comment by jen20** [comment]
@ahmetalpbalkan done, for now I've added it to the `.travis.yml` file instead.

**Pull request merged**